acorn/src/statement.js

```diff
@@ -666,7 +673,9 @@ pp.parseClassElement = function(constructorAllowsSuper) {
     // Couldn't move this check into the 'parseClassMethod' method for backward compatibility.
     if (isConstructor && kind !== "method") this.raise(node.key.start, "Constructor can't have get/set modifier")
     node.kind = isConstructor ? "constructor" : kind
-    this.parseClassMethod(node, isGenerator, isAsync, allowsDirectSuper)
+    if (isPrivate) {
+      this.parseClassPrivateMethod(node, isGenerator, isAsync, allowsDirectSuper)
```

Comment:
You should be able to just override `parseClassMethod` and check `node.name.type`, calling through to `super` to use the default parsing logic if you need to, so I don't think an additional method is needed here.